vite.js

```diff
@@ -4,91 +4,111 @@ const processCSS = require('./src/process-css');
 const babelPlugin = require('./babel.js');
 const NAME = require('./package.json').name;
 
-const VIRTUAL_MODULE_NAME = '@virtual-style9';
+const VIRTUAL_MODULE_NAME = '\0plugin-style9:virtual-module';
+const VIRTUAL_CSS_NAME = VIRTUAL_MODULE_NAME + '.css';
 
+async function transformStyle9(
+  code,
+  id,
+  { parserOptions, restOptions, server, cssModules }
+) {
+  const res = await babel.transformAsync(code, {
+    plugins: [[babelPlugin, restOptions]],
+    filename: id,
+    sourceMaps: true,
+    parserOpts: parserOptions,
+    babelrc: false
+  });
+  if (!res) {
+    return;
+  }
+  const { code: _code, map, metadata } = res;
+  const output = [_code];
+  const css = metadata.style9;
+  if (css) {
+    cssModules.set(id, css);
+    output.unshift(`import '${VIRTUAL_MODULE_NAME}'`);
+    if (server) {
+      const { moduleGraph } = server;
+      const virtualModule = moduleGraph.getModuleById(VIRTUAL_CSS_NAME);
+      if (virtualModule) {
+        const { code: cssStr } = virtualModule.transformResult;
+        const latest = cssStr
+          .split('\n')
+          .map(str => {
+            if (str.startsWith('const __vite__css')) {
```

Comment:
This depends on a line starting with this exact string

Reply:
Yes. Vite converts css modules to js modules. According https://github.com/vitejs/vite/blob/main/packages/vite/src/node/plugins/css.ts#L398-L411

Reply:
Like I said, that's an implementation detail that's not guaranteed to be stable across versions

Reply:
okay. I'll use another way